src/lib/routing/router.ts

```diff
@@ -20,21 +20,47 @@ const defaultErrorHandler = (auto405) => async (err: Error, req: EReq, res: ERes
   res.withStatus(500).json({ error: err.message });
 }
 
+const preflightHandler = (autoCorsPreflight: AutoCorsPreflightOptions) => async (req: EReq, res: ERes) => {
+  if(autoCorsPreflight.trustedOrigins.length===0){
+    return;
+  }
+  let originHeaderValue :string | null = null;
+  if(autoCorsPreflight.trustedOrigins.length===1 && autoCorsPreflight.trustedOrigins[0] === '*'){
+    originHeaderValue = '*';
+  }else{
+    const origin = req.headers.get("origin");
+    if (autoCorsPreflight.trustedOrigins.includes(origin)){
+      originHeaderValue = origin;
+    }
+  }
+  if(originHeaderValue==null){
+    return;
+  }
+  res.set({
+    'access-control-allow-origin': originHeaderValue,
+    'access-control-allow-methods': 'GET, PUT, POST, DELETE, PATCH, OPTIONS'
+  })
+  return res.sendStatus(200);
+}
+
 export class Router {
   requestHandlers: Array<RequestHandler> = [];
   errorHandlers: Array<ErrorMiddleware> = [];
   config: EConfig = {
     parseCookie: true,
     auto405: true,
     extractRequestParameters: true,
-    autoContentType: false
+    autoContentType: false,
   };
 
   constructor(config?: EConfig) {
     this.config = {
       ...this.config,
       ...config
     }
+    if(this.config.autoCorsPreflight!= null){
```

Comment:
You'll want a `!==` here.  Generally speaking there's never a good use for `!=` or `==`.  If you just want to test for falsyness (which I think would be fine here), cast it to a boolean:

```suggestion
    if (!this.config.autoCorsPreflight) {
```

This makes it clear that you never intended to check for equality to `null` specifically.